torch/distributed/elastic/rendezvous/dynamic_rendezvous.py

```diff
@@ -1188,25 +1188,19 @@ def next_rendezvous(self) -> RendezvousInfo:
                 bootstrap_store_info,
             )
 
+        # This will only be hit when TCPStore sharing is enabled.
         if self._bootstrap_store_info is None:
-            if isinstance(self._store, dist.TCPStore):
-                addr = self._store.host
-                port = self._store.port
-                self._bootstrap_store_info = RendezvousStoreInfo(
-                    master_addr=addr, master_port=port
-                )
-                if rank == 0:
-                    self._shared_tcp_store_server = self._store
-            else:
-                # If the store is not type of TCPStore start TCPStore server, which requries
-                # bootstrapping info across ranks
-                self._bootstrap_store_info = RendezvousStoreInfo.build(
-                    rank, store, local_addr=self._this_node.addr
+            # To ensure no race for each retry. We always create a TCPStore and this requries
+            # bootstrapping info across ranks
+            self._bootstrap_store_info = RendezvousStoreInfo.build(
+                rank, store, local_addr=self._this_node.addr
+            )
```

Comment:
Can we move this after the rank0 check and instead pass in the port from `_shared_tcp_store_server` on rank0 to be broadcasted?